### PR TITLE
misc cleanups

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: src/basicio.cpp [hunk 1]
```diff
@@ -395,13 +395,11 @@
         EXV_WARNING << Error(ErrorCode::kerCallFailed, pf, strError(), "::stat") << "\n";
 #endif
       }
-      if (statOk && origStMode != buf2.st_mode) {
-        // Set original file permissions
-        if (::chmod(pf, origStMode) == -1) {
+      // Set original file permissions
+      if (statOk && origStMode != buf2.st_mode && ::chmod(pf, origStMode) == -1) {
```
[the remark this answers is not in the document]
Comment (neheb) [reply src/basicio.cpp:399]: Hahaha no.

**Comment by neheb** [comment]
```
 Error:  Failed to execute: D:\a\exiv2\exiv2\build\bin\iotest s0 s1 s2 http://127.0.0.1:12760/table.jpg
Error:  The asserted return code is [0], but got 1
```

I've seen this before. Transient error.

**Review by piponazo — APPROVED** [review]
LGTM! Nice set of changes 👏

**Comment by piponazo** [comment]
> ```
>  Error:  Failed to execute: D:\a\exiv2\exiv2\build\bin\iotest s0 s1 s2 http://127.0.0.1:12760/table.jpg
> Error:  The asserted return code is [0], but got 1
> ```
> 
> I've seen this before. Transient error.

If this is a flaky test, I would vote to comment it out an create a ticket to investigate what's happening there. In that way it will not create confusion.

**Pull request merged**